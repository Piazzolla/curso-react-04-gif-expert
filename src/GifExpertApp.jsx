import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Bandoneon']);
    const onAddCategory = (newCategory) => {
        // categories.push('Hola mundo');
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
    }

    return (
        <>

            {/*   Titulo    */}
            <h1>GifExpertApp</h1>

            {/*   Input    */}
            <AddCategory
                onNewCategory={onAddCategory}
            />

            {/*   Listado de Gif    */}
                {categories.map(category => (
                    <GifGrid key={ category } category={ category } />
                ))}


            {/*   Gif item    */}


        </>
    )
}
