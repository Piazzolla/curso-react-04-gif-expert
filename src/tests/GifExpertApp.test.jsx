import { render, screen } from "@testing-library/react";
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => { 
    test('Debe matchear con el snapshot', () => { 
        const {container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('debe verificar que las categorias no se repitan', () => { 
        const {container } = render(<GifExpertApp />);
        
      //  TODO...


     });

     test('Verifico que este el titulo', () => { 
        const title = 'GifExpertApp';
        const {container } = render(<GifExpertApp />);
        expect( screen.getByText( title )).toBeTruthy();
     });
 })