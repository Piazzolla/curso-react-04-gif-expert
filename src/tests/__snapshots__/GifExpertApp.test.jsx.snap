// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe matchear con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Bandoneon
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
